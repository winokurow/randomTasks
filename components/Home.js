import React, { useState, useCallback } from 'react';
import { useDispatch, useSelector, shallowEqual } from "react-redux";
import { StyleSheet, Text, View, Button, Icon, TouchableOpacity } from 'react-native';
import { Header, Left } from 'react-native-elements';
import { useNavigation, DrawerActions } from '@react-navigation/native'

const  Home = () => {
  const [taskName, setTaskName] = useState('test')
  let tasks = useSelector(state => state.subjects.all_subjects, shallowEqual)
  const navigation = useNavigation();

  const chooseTask = () => {    
    // First, we loop the main dataset to count up the total weight. We're starting the counter at one because the upper boundary of Math.random() is exclusive.
    let total = 1;
    for (let i = 0; i < tasks.length; ++i) {
        total += tasks[i].taskWeight;
    }
    console.log(total)
    // Total in hand, we can now pick a random value akin to our
    // random index from before.
    const threshold = Math.floor(Math.random() * total);
    console.log(threshold)
    // Now we just need to loop through the main data one more time
    // until we discover which value would live within this
    // particular threshold. We need to keep a running count of
    // weights as we go, so let's just reuse the "total" variable
    // since it was already declared.
    total = 0;
    for (let i = 0; i < tasks.length; ++i) {
      // Add the weight to our running total.
      total += tasks[i].taskWeight;

      // If this value falls within the threshold, we're done!
      if (total >= threshold) {
        setTaskName(tasks[i].taskName)
        break
    }
}

    
  }

  return (
    <View style={styles.container}>
      <Header leftComponent={{ icon: 'menu', color: '#fff', onPress: () => navigation.dispatch(DrawerActions.toggleDrawer()), }}></Header>
           
      <Text style={styles.task}>{taskName}</Text>
      <Button onPress={chooseTask} title='Get Task' style={styles.button}/>
    </View>
  );
}
 
 

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'flex-start',
    backgroundColor: '#a3bcc7cc7',
  },
  task: {
      paddingBottom: 20,
      fontSize: 25,
      fontWeight: 'bold',
      paddingLeft: 10,
      borderBottomWidth: 1
  },
  button: {
    fontSize: 15
}
});


export default Home;

