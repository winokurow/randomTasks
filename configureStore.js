import { createStore } from 'redux'
import { persistStore, persistReducer } from 'redux-persist'
import ExpoFileSystemStorage from "redux-persist-expo-filesystem"
import subjectsReducer from './components/SubjectsReducer'

const persistConfig = {
    key: 'subjectsReducer',
    storage: ExpoFileSystemStorage,
}

const persistedReducer = persistReducer(persistConfig, subjectsReducer)




export default () => {

  let store = createStore(persistedReducer)
  let persistor = persistStore(store)
  return { store, persistor }
}