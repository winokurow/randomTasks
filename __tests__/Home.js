import React from 'react';
import { act, create, renderer } from 'react-test-renderer';
import Home from '../components/Home';
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const ReduxProvider = ({ children, reduxStore }) => (
  <Provider store={reduxStore}>{children}</Provider>
)
jest.mock('@react-navigation/native');

describe('Home Component', () => {
  subjects = {
    'all_subjects':[{'taskName':'test', 'taskWeight':134}]
  }
  const initialState = {'subjects': subjects}
  const mockStore = configureStore()

  xit('App renders without crashing', async () => {
    store = mockStore(initialState)
    let tree
    await act(async () => {



      tree = create(<Provider store={store}><Home></Home></Provider>);
    })
    await act(async () => {
      expect(tree.toJSON()).toMatchSnapshot();

      console.log('*************' + tree);
      expect(tree).toBeTruthy();
    })
})
})