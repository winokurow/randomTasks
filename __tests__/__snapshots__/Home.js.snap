// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component App renders without crashing 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#a3bcc7",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#2089dc",
        "borderBottomColor": "#f2f2f2",
        "borderBottomWidth": 0.5,
        "flexDirection": "row",
        "height": 64,
        "justifyContent": "space-between",
        "paddingHorizontal": 10,
        "paddingTop": 20,
      }
    }
    testID="headerContainer"
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "justifyContent": "center",
              }
            }
          >
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "#fff",
                    "fontSize": 24,
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                  Object {
                    "fontFamily": "material",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              testID="iconIcon"
            >
              î—’
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 3,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "borderBottomWidth": 1,
        "fontSize": 25,
        "fontWeight": "bold",
        "paddingBottom": 20,
        "paddingLeft": 10,
      }
    }
  >
    test
  </Text>
  <View
    accessibilityRole="button"
    accessibilityState={Object {}}
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Get Task
      </Text>
    </View>
  </View>
</View>
`;
